pkg/core/combat/combat.go

```diff
@@ -60,20 +62,32 @@ func (h *Handler) Target(i int) Target {
 	return h.targets[i]
 }
 
+func (h *Handler) cachePrimaryTarget() {
```

Comment:
Although I've added it here, I don't think caching will necessary give a performance increase.

Reply:
I haven't looked at the PR in detail yet but if you are caching then will also need to call cache function when you set player pos